Doc/library/collections.abc.rst

```diff
@@ -272,6 +272,12 @@ Collections Abstract Base Classes -- Detailed Descriptions
       The index() method added support for *stop* and *start*
       arguments.
 
+   .. deprecated:: 3.12
```

Comment:
Do you have a removal version in mind, or would you prefer to keep it open ended?

In any case, please could you also document this deprecation at https://docs.python.org/3.12/whatsnew/3.12.html#deprecated or its appropriate subsection?

Reply:
Thanks, updated the role and added a What's New entry

Close ended is preferable in this case; I already use warnings._deprecated to ensure removal or deferral come 3.14

Reply:
Thanks, then we can stick it directly at https://docs.python.org/3.12/whatsnew/3.12.html#pending-removal-in-python-3-14